internal/services/datafactory/data_factory_linked_service_sftp_resource.go

```diff
@@ -104,33 +121,124 @@ func resourceDataFactoryLinkedServiceSFTP() *pluginsdk.Resource {
+				AtLeastOneOf:  []string{"password", "key_vault_password", "private_key_content_base64", "key_vault_private_key_content_base64", "private_key_path"},
+			},
+
+			"key_vault_password": {
+				Type:          pluginsdk.TypeList,
+				Optional:      true,
+				ConflictsWith: []string{"private_key_content_base64", "private_key_path", "private_key_passphrase"},
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"linked_service_name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"secret_name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
 				},
 			},
 
-			"additional_properties": {
-				Type:     pluginsdk.TypeMap,
-				Optional: true,
-				Elem: &pluginsdk.Schema{
-					Type: pluginsdk.TypeString,
+			"private_key_content_base64": {
+				Type:          pluginsdk.TypeString,
+				Optional:      true,
+				Sensitive:     true,
+				ValidateFunc:  validation.StringIsBase64,
+				ConflictsWith: []string{"private_key_path", "password", "key_vault_password", "key_vault_private_key_content_base64"},
+			},
+
+			"key_vault_private_key_content_base64": {
+				Type:          pluginsdk.TypeList,
+				Optional:      true,
+				ConflictsWith: []string{"private_key_path", "password", "key_vault_password", "private_key_content_base64"},
+				MaxItems:      1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"linked_service_name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"secret_name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
 				},
 			},
+
+			"private_key_path": {
+				Type:          pluginsdk.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"private_key_content_base64", "key_vault_private_key_content_base64", "key_vault_password", "password"},
+			},
+
+			"skip_host_key_validation": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+			},
 		},
+		CustomizeDiff: pluginsdk.CustomDiffWithAll(
+			pluginsdk.CustomizeDiffShim(func(ctx context.Context, d *pluginsdk.ResourceDiff, v interface{}) error {
+				for authType, reqs := range map[string][]string{
+					string(datafactory.SftpAuthenticationTypeSSHPublicKey): {"private_key_content_base64", "key_vault_private_key_content_base64", "private_key_path"},
+					string(datafactory.SftpAuthenticationTypeBasic):        {"password", "key_vault_password"},
+				} {
+					for _, x := range reqs {
+						if _, ok := d.GetOk(x); ok {
+							if d.Get("authentication_type").(string) != authType {
+								return fmt.Errorf("`authentication_type` must be `%s` when `%s` is defined", authType, x)
+							}
+						}
+					}
+				}
```

Comment:
I think we could improve the readability here by doing 2 things:

1. Define a variable for the map instead of defining this inline with the loop declaration
2. Change the structure of the map to get rid of the nested loop

Something like so:

```suggestion
				authCombinations := map[string]string{
					"private_key_content_base64":           string(datafactory.SftpAuthenticationTypeSSHPublicKey),
					"key_vault_private_key_content_base64": string(datafactory.SftpAuthenticationTypeSSHPublicKey),
					"private_key_path":                     string(datafactory.SftpAuthenticationTypeSSHPublicKey),
					"password":                             string(datafactory.SftpAuthenticationTypeBasic),
					"key_vault_password":                   string(datafactory.SftpAuthenticationTypeBasic),
				}

				for keyType, authType := range authCombinations {
					if _, ok := d.GetOk(keyType); ok {
						if d.Get("authentication_type").(string) != authType {
							return fmt.Errorf("`authentication_type` must be `%s` when `%s` is defined", authType, keyType)
						}
					}
				}
```

A few more lines, but I think it's easier to read if we ever have to come back to this, wdyt?

Reply:
I'm fine either way, I just wanted to avoid a only-used-one-time variable. If you think it's justifiable in this case, I'm ok with it.